doc/kernel/usermode/usermode.rst

```diff
@@ -9,6 +9,176 @@ are defined, and how memory may be managed to support user mode threads.
 For details on creating threads that run in user mode, please see
 :ref:`lifecycle_v2`.
 
+Threat Model
+============
+
+User mode threads are considered to be untrusted by Zephyr and are therefore
+isolated from other user mode threads and from the kernel. A flawed or
+malicious user mode thread cannot leak or modify the private data/resources
+of another thread or the kernel, and cannot interfere with or
+control another user mode thread or the kernel.
+
+Example use-cases of Zephyr's user mode features:
+
+- The kernel can protect against unintentional programming errors which could
```

Comment:
Would memory domain misconfiguration be classified under "unintended programming error"? If so, I'd change this to say "the kernel can protect against _many_ unintentional (...)"

Reply:
user threads can't change memory domain config. no programming errors that execute in supervisor mode can be prevented.

Reply:
Added the "many" verbiage though.